engine/flink/components/base/src/main/scala/pl/touk/nussknacker/engine/flink/util/transformer/aggregate/transformers.scala

```diff
@@ -85,21 +86,22 @@ object transformers {
           val keyedStream = start
             .groupByWithValue(groupBy, aggregateBy)
           val aggregatingFunction = new UnwrappingAggregateFunction[AnyRef](aggregator, aggregateBy.returnType, identity)
-          val windowDefinition = TumblingEventTimeWindows.of(Time.milliseconds(windowLength.toMillis))
+          val offsetMillis = windowOffsetProvider.offset(windowLength).toMillis
+          val windowDefinition = TumblingEventTimeWindows.of(Time.milliseconds(windowLength.toMillis), Time.milliseconds(offsetMillis))
 
           (tumblingWindowTrigger match {
             case TumblingWindowTrigger.OnEvent =>
               keyedStream
                 .eventTriggerWindow(windowDefinition, typeInfos, aggregatingFunction, EventTimeTrigger.create())
             case TumblingWindowTrigger.OnEnd =>
               keyedStream
-                 .window(windowDefinition)
-                 .aggregate(aggregatingFunction,
-                   EnrichingWithKeyFunction(fctx), typeInfos.storedTypeInfo, typeInfos.returnTypeInfo, typeInfos.returnedValueTypeInfo)
+                .window(windowDefinition)
+                .aggregate(aggregatingFunction,
+                  EnrichingWithKeyFunction(fctx), typeInfos.storedTypeInfo, typeInfos.returnTypeInfo, typeInfos.returnedValueTypeInfo)
             case TumblingWindowTrigger.OnEndWithExtraWindow =>
               keyedStream
-                 //TODO: alignment??
-                 .process(new EmitExtraWindowWhenNoDataTumblingAggregatorFunction[SortedMap](aggregator, windowLength.toMillis, nodeId, aggregateBy.returnType, typeInfos.storedTypeInfo, fctx.convertToEngineRuntimeContext))
+                //TODO: alignment??
```

Comment:
Is this still relevant? Maybe alignemnt was meant as alignment to time zone?

Reply:
I think it is still relevant